cmd/soroban-rpc/internal/db/ledgerentry.go

```diff
@@ -355,7 +362,8 @@ func (r ledgerEntryReader) NewCachedTx(ctx context.Context) (LedgerEntryReadTx,
 	// in between, updating the cache.
 	r.db.cache.RLock()
 	defer r.db.cache.RUnlock()
-	if err := txSession.BeginTx(ctx, &sql.TxOptions{ReadOnly: true}); err != nil {
+	txSession, err := r.newBaseTx(ctx)
```

Comment:
I think we can get rid of the `newBase()` function and instead you can implement `NewCachedTx()` by calling `NewTx()` and then setting the `ledgerEntryCacheReadTx` and `stmtCache` fields on the *ledgerEntryReadTx instance before returning it